assembl/tasks/piwik.py

```diff
@@ -0,0 +1,321 @@
+        raise requests.ConnectionError()
+
+    content = result.json() # returns something like [{"login":"aaa","email":"aaa@aaa.com"}] or {"result":"error","message":"L'utilisateur 'aaa@aaa.com' est inexistant."}
+    # print "piwik_UsersManager_getUserByEmail", content
+    if not content:
+        raise requests.ConnectionError()
+
+    if "result" in content and content["result"] == "error":
+        return False
+    elif not isinstance(content, list):
+        raise requests.ConnectionError()
+    else:
+        return content
+
+
+def piwik_UsersManager_addUser(piwik_url, piwik_api_token, userLogin, password, email, alias=''):
+    params = {
+        "module": "API",
+        "format": "JSON",
+        "token_auth": piwik_api_token
+    }
+    params["method"] = "UsersManager.addUser"
+    params["userLogin"] = userLogin # Piwik has two different fields for login and email, but a user can have the same value as login and email
+    params["password"] = password
+    params["email"] = email
+    if ( alias ):
+        params["alias"] = alias
+    result = requests.get(piwik_url, params=params, timeout=15)
+    if result.status_code != 200:
+        raise requests.ConnectionError()
+
+    content = result.json()
+    # print "piwik_UsersManager_addUser", content
+    if not content:
+        raise requests.ConnectionError()
+
+    user_added = ("result" in content and content["result"] == "success")
+    return user_added
+
+
+def piwik_SitesManager_getSitesIdFromSiteUrl(piwik_url, piwik_api_token, url):
+    params = {
+        "module": "API",
+        "format": "JSON",
+        "token_auth": piwik_api_token
+    }
+    params["method"] = "SitesManager.getSitesIdFromSiteUrl"
+    params["url"] = url
+    result = requests.get(piwik_url, params=params, timeout=15)
+
+    if result.status_code != 200:
+        raise requests.ConnectionError()
+    content = result.json() # Content should be either an empty array, or an array like [{"idsite":"44"}]
+    # print "piwik_SitesManager_getSitesIdFromSiteUrl", content
+    if not isinstance(content, list):
+        raise requests.ConnectionError()
+
+    return content
+
+
+def piwik_SitesManager_addSite(piwik_url, piwik_api_token, siteName, urls, ecommerce = '', siteSearch = '', searchKeywordParameters = '', searchCategoryParameters = '', excludedIps = '', excludedQueryParameters = '', timezone = '', currency = '', group = '', startDate = '', excludedUserAgents = '', keepURLFragments = '', param_type = '', settings = '', excludeUnknownUrls = ''):
+    """Returns the Piwik id_site of the created website
+    Warning: A new Piwik site is created everytime this method is called, even if another Piwik website already exists with the same name and URLs"""
+    params = {
+        "module": "API",
+        "format": "JSON",
+        "token_auth": piwik_api_token
+    }
+    params["method"] = "SitesManager.addSite"
+    params["siteName"] = siteName
+    params["urls"] = urls
+    if ( ecommerce ):
```

Comment:
Not bad! As a one-liner:
params.extend({k: v for (k,v) in kwargs.iteritems() if k in extras})